src/App.tsx

```diff
@@ -1,61 +1,52 @@
+import { useState } from 'react';
+
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { getNewId, getTodosWithUsers } from './helpers';
+import { AddPostForm } from './components/AddPostForm';
+import { Todo } from './types/interfaces';
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [users] = useState(usersFromServer);
+
+  const todosWithUsers = getTodosWithUsers(todos, users);
+
+  const addTodo = (
+    todoTitle: string,
+    userId: number,
+    completed: boolean,
+  ) => {
+    setTodos((currentTodos) => {
+      const newTodo: Todo = {
+        id: getNewId(currentTodos),
+        title: todoTitle,
+        completed,
+        userId,
+      };
+
+      return [
+        ...currentTodos,
+        newTodo,
+      ];
+    });
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <AddPostForm
+        addTodo={addTodo}
+        users={users}
+      />
+
+      <TodoList
+        todos={todosWithUsers}
+      />
```

Comment:
```suggestion
      <TodoList todos={todosWithUsers} />
```
You can write it in one line if there are < 3 props